cmd/genkubedocs/gen_kube_docs.go

```diff
@@ -34,6 +35,7 @@ import (
 )
 
 func main() {
+	logger := klog.Background()
```

Comment:
It would be better to pass context instead of logger, the same way as it's done for cmd/kube-proxy and cmd/scheduler. Context is more functional and preferable in my opinion.

Reply:
I appreciate your comment.
I will check them and fix all parts (in this PR) to pass context.

Reply:
@bart0sh 
I replaced from logger to context with https://github.com/kubernetes/kubernetes/pull/127887/commits/32b3982aba6f1a78465ca045e9deda55dc7f6dc5 
I believe it's better now, but what do you think?

Reply:
It's definitely better now. Can you pass the context down the stack a little bit and get rid of at least some of the `context.Background` and `context.TODO` calls in the pkg/kubelet/kubelet.go? I know, it's a bit out of scope of this PR, but it would be great to pass the context there and show SIG-Node developers how to use it.

I'm not asking to change every call of `context.TODO` and `context.Background`, just in a couple of obvious places like [Kubelet.Run](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1598), [NewMainKubelet](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L376) etc.

Reply:
@bart0sh 
I created [PR#128027](https://github.com/kubernetes/kubernetes/pull/128027) for further work you mentioned, please confirm if it's what you wanted😊
I will also work on Support Contextual logging in pkg/kubelet soon👍

Reply:
It would be great if you simply add those changes to this PR as a separate commit(s). It will look more understandable if you do it here, at least in my opinion.

Reply:
Thank you, I understand now👍
@rphillips says another open PR adds ctx to pkg/kubelet: cf.[#128008](https://github.com/kubernetes/kubernetes/pull/128008).
So, unfortunately, this PR should limit the scope to cmd/kubelet.

Reply:
That pr has different goal, I believe, but yeah, let's keep this as is.
/lgtm

Reply:
It was a good opportunity to take a look at the pkg/kubelet codes. 
Thanks for your suggestions😊

Reply:
/lgtm